packages/workers-shared/asset-worker/tests/handler.test.ts

```diff
@@ -0,0 +1,128 @@
+import { vi } from "vitest";
+import { handleRequest } from "../src/handler";
+import type { AssetConfig } from "../../utils/types";
+
+describe("[Asset Worker] `handleRequest`", () => {
+	it("attaches ETag headers to responses", async () => {
+		const configuration: Required<AssetConfig> = {
+			html_handling: "none",
+			not_found_handling: "none",
+		};
+		const eTag = "some-etag";
+		const exists = vi.fn().mockReturnValue(eTag);
+		const getByETag = vi.fn().mockReturnValue({
+			readableStream: new ReadableStream(),
+			contentType: "text/html",
+		});
+
+		const response = await handleRequest(
+			new Request("https://example.com/"),
+			configuration,
+			exists,
+			getByETag,
+			false
+		);
+
+		expect(response.headers.get("ETag")).toBe(`"${eTag}"`);
+	});
+
+	it("suppresses 304 Not Modified responses when `skipIfNoneMatchHeaderHandling` is set to true", async () => {
+		const configuration: Required<AssetConfig> = {
+			html_handling: "none",
+			not_found_handling: "none",
+		};
+		const eTag = "some-etag";
+		const exists = vi.fn().mockReturnValue(eTag);
+		const getByETag = vi.fn().mockReturnValue({
+			readableStream: new ReadableStream(),
+			contentType: "text/html",
+		});
+
+		const response = await handleRequest(
+			new Request("https://example.com/", {
+				headers: { "If-None-Match": eTag },
```

Comment:
I think the `eTag` might need to be included in quotes here.

```suggestion
				headers: { "If-None-Match": `"${eTag}"` },
```

Reply:
Good catch, thank you!